doc/OnlineDocs/installation.rst

```diff
@@ -42,3 +36,18 @@ the following in a shell:
 ::
 
    pip install pyomo
+
+
+Conditional Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Extensions to Pyomo, and many of the contributions in `pyomo.contrib`,
+also have conditional dependencies on a variety of third-party Python
+packages including but not limited to: numpy, scipy, sympy, networkx,
+openpxl, pyodbc, xlrd, pandas, matplotlib, pymysql, pyro4, and
+pint. 
```

Comment:
Suggestion: "Each extension should generate an error message for missing dependencies upon use."

Reply:
Thanks, that's a good suggestion.